Sources/Controllers/MessagesViewController.swift

```diff
@@ -141,6 +142,20 @@ open class MessagesViewController: UIViewController {
             let trailing = messagesCollectionView.trailingAnchor.constraint(equalTo: view.trailingAnchor)
             NSLayoutConstraint.activate([top, bottom, trailing, leading])
         }
+        adjustScrollViewInset()
+    }
+    
+    @objc
+    private func adjustScrollViewInset() {
+        if #available(iOS 11.0, *) {
```

Comment:
I guess we can't `!` this statement?

Nonetheless, this PR looks promising. I'd just like to test it against iOS 9-11 and the iPhone X. Also both opaque and translucent navigation bars

Reply:
Ya you can't ! and `#available` because I think `#available` is a compiler operation (it swift is similar to C/C++